### PR TITLE
Correlation ids

### PR DESCRIPTION
**Pull request by janwiberg-dfds**
Add x-correlation-id to http request/responses and carry that into domain events, and thus kafka

### CODE REVIEW

File: src/CapabilityService.WebApi/Startup.cs [hunk 1]
```diff
@@ -36,12 +41,28 @@ public Startup(IConfiguration configuration)
 
         public void ConfigureServices(IServiceCollection services)
         {
-            services
-                .AddMvc()
-                .SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Latest);
```
Comment (jakob-nielsen-dfds) [inline src/CapabilityService.WebApi/Startup.cs]: Isn't it better to be explicit about version, eg. the one the solution is using (2.2)?

File: src/CapabilityService.WebApi/Program.cs [hunk 2]
```diff
@@ -8,19 +8,33 @@
...
-            Log.Logger = new LoggerConfiguration()
+            bool.TryParse(Environment.GetEnvironmentVariable("CAPABILITY_SERVICE_HUMAN_LOG"), out var humanLog);
+
+            var logcfg = new LoggerConfiguration()
                 .MinimumLevel.Information()
                 .MinimumLevel.Override("Microsoft", LogEventLevel.Warning)
-                .Enrich.FromLogContext()
-                .WriteTo.Console(new CompactJsonFormatter())
-                .CreateLogger();
+                .MinimumLevel.Override(source: "Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker", LogEventLevel.Information)
```
Comment (jakob-nielsen-dfds) [inline src/CapabilityService.WebApi/Program.cs]: Build error line 24 and 25: Named arguments must appear after all fixed arguments.

**Review by jakob-nielsen-dfds — CHANGES_REQUESTED** [review]

File: src/CapabilityService.WebApi/Program.cs (hunk 2 above)
Comment (janwiberg-dfds) [reply src/CapabilityService.WebApi/Program.cs]: Hrm, it builds just fine here. Which version are you on?

Comment (jakob-nielsen-dfds) [reply src/CapabilityService.WebApi/Program.cs]: Using rider or _dotnet build_?

Version info:
_dotnet --info_
.NET Core SDK (reflecting any global.json):
 Version:   2.2.105
 Commit:    7cecb35b92

Runtime Environment:
 OS Name:     manjaro
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /opt/dotnet/sdk/2.2.105/

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6

.NET Core SDKs installed:
  2.1.402 [/opt/dotnet/sdk]
  2.1.403 [/opt/dotnet/sdk]
  2.1.500 [/opt/dotnet/sdk]
  2.2.105 [/opt/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.4 [/opt/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [/opt/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [/opt/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.4 [/opt/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [/opt/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [/opt/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.4 [/opt/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [/opt/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [/opt/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [/opt/dotnet/shared/Microsoft.NETCore.App]

Comment (SEQUOIIA) [reply src/CapabilityService.WebApi/Program.cs]: Adding an explicit C# version to the .csproj allows for compilation, e.g.
```
  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <RootNamespace>DFDS.CapabilityService.WebApi</RootNamespace>
    <AssemblyName>CapabilityService.WebApi</AssemblyName>
    <LangVersion>7.2</LangVersion>
  </PropertyGroup>
```
```<LangVersion>7.2</LangVersion>```

Comment (janwiberg-dfds) [reply src/CapabilityService.WebApi/Program.cs]: Nice find. Still not sure why though.

**Review by jakob-nielsen-dfds — APPROVED** [review]

**Pull request merged**